Update public user

ryanrath opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Until we completely remove role usage we need to create a 'Public' role as
well as an association to this role for the Public User. As such, we need to ensure that the Public role exists and that the Public User has a relation to it. In addition, the enumAllAvailableRoles function needs to be updated so that it can take this new role into account.

## Motivation and Context
Without these changes the Public User's 'activeRole' returns the user role. This leads to unexpected behavior when viewing the Usage tab ( i.e. there are a number of Realms that are missing when logged in as the Public User ).

## Tests performed
Manual Tests performed: 

Test 1:
  - Navigate to XDMoD ( do not login )
  - Click the Usage tab
  - Double check that the 'Accounts' and 'Allocations' realm are visible / enabled.

Test 2: 
  - Navigate to XDMoD ( login as a 'usr' )
  - Click the Usage tab
  - Double check that the 'Accounts' and 'Allocations' realms are not visible

Test 3:
  - Navigate to XDMoD ( login as a 'cd' )
  - Click the Usage tab
  - Double check that the 'Accounts' and 'Allocations' realms are visible.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

smgallo commented on configuration/etl/etl_sql.d/acls/xdmod/create_public_user.sql:
```diff
@@ -30,11 +30,43 @@ FROM (
+    true as is_primary,
+    true as is_active
+  FROM Users u, Roles r
+  WHERE     BINARY u.username = BINARY 'Public User'
```
Should this also be 'Public' as above at line 33?

ryanrath replied:
That it should, should be changed in the latest commit.

ryanrath commented:
Yup, it most definitely should be made against the 7.0 branch. I've updated the Target to 7.0

jpwhite4 commented:
Looks like this change has broken Open XDMoD.

ryanrath commented:
I'm looking into why that test failed. I had executed the tests on a local docker image w/ this branch and they all passed ( hence submitting the pr ). I'm spinning up a new container to double check things now.

jpwhite4 commented on configuration/etl/etl_sql.d/acls/xdmod/create_public_user.sql:
```diff
@@ -30,11 +30,43 @@ FROM (
         a.acl_id AS acl_id
      FROM Users u, acls a
-     WHERE BINARY u.username = BINARY 'Public User' AND
+     WHERE BINARY u.username = BINARY 'Public' AND
```
how come the username string is 'Public' here and then 'Public User' on line 65?

smgallo commented on configuration/etl/etl_sql.d/acls/xdmod/create_public_user.sql:
```diff
@@ -30,7 +30,7 @@ FROM (
         a.acl_id AS acl_id
      FROM Users u, acls a
-     WHERE BINARY u.username = BINARY 'Public' AND
+     WHERE BINARY u.username = BINARY 'Public User' AND
```
Now I'm confused. In a message from 10/2/17 you said that the value should be "Public" and changed it in commit 
0d3c6c9 but then changed one back again in commit 5ff1388. So should they both be "Public" or "Public User"?

ryanrath replied:
They should both be 'Public User'. At this point I'm not sure what temporary insanity came over me when I changed it to 'Public'. I'm going to conduct another test on a fresh container to make sure that everything is copacetic.

ryanrath replied:
Ok, test completed on the centos7-xdmod6.6.0:version6 docker container. Everything looks to have sorted itself out. I will also run a test w/ the centos7-xdmod-supremm6.0:version3 ( I'll also check to see if there's a newer version ) container.

ryanrath replied:
Aaaand, tested with the centos7-xdmod-supremm6.6.0:version3 container. No errors w/ integration or component tests.

ryanrath commented:
@smgallo @jpwhite4 @chakrabortyr 
Alrighty all, I believe I've addressed the comments thus far. The tests are passing, I've added a test to detect the issue I ran into during testing / during the deployment to xdmod-staging. If there's anything that I've missed please let me know.

Thanks!

jpwhite4 commented on open_xdmod/modules/xdmod/integration_tests/lib/Controllers/UsageExplorerTest.php:
```diff
@@ -316,4 +318,63 @@ public function exportDataProvider()
+     **/
+    public function getRealms()
+    {
+        $config = Config::factory();
```
This won;t work

jpwhite4 commented:
Don't forget to:
1) test this on XSEDE instance and confirm that the correct realms appear
2) also merge this change into xdmod 7.1
3) change the test config to just expect what you want it to (i.e. Jobs realm only).

ryanrath commented:
Tested against an XSEDE install w/ App Kernels & SUPREMM, the latest commit ( a minor change to acl-config ) was the only issue I found. As long as the automated tests pass then I'll go ahead and merge
